File: classes/003/projects/c++/calculator/calculator.cpp
// Standard Input / Output Streams Library
// Has the funcitons cout and cin
// the command include adds functionality to your program
// iostream is part of the standard c++ library (STL), which has a lot
// of other functionalities that you can leverage.
// For more on STL, check, https://www.cplusplus.com/reference
#include <iostream>

int main() {
  // Variable declarations
  // Variables are used to store information
  // int means that this variable is of type integer (number)
  // = 0 is saying to the variavel to store 0 as its current value.
  // Always a good practice to initialize your variables, that means
  // assigning a value to the recently created variable.
  int first  = 0;
  int second = 0;

  // cout prints the text to the console
  // << is the direction from where the information will flow
  // in this line it's passing the text to cout
  // std:: is the namespace where the function is,
  // namespaces are used to organize code and avoid name conflicts
  std::cout << "Enter the first positive integer : ";
  // cin waits for the user input
  // >> means that cin will give information to the variable
  // cin will stop when the user hits enter
  std::cin >> first;
  std::cout << "Enter the second positive integer : ";
  std::cin >> second;
  // cout accepts more than on stream (<<) you can use multiple on your
  // calls.
  // first + second is a command to sum variables of the same type,
  //  in this case, int + int, will result in a new integer
  // you can assign the result of a sum to a variable
  // int a = b + c;
  // endl is the end of the line, is the same as hiting enter on your
  // keyboard when typing a text.
  std::cout << "The sum is : " << first + second << std::endl;

  // return is a concept that we will study in future classes,
  // you don't need to worry about it now.
  return 0;
}

File: classes/003/projects/python/calculator/calculator.py
# first is a variable that can store values
# python is dynamically typed, this means that you don't have to define
# its type on declaration.

# for naming, there are some rules that you can check here,
# https://www.w3schools.com/python/gloss_python_variable_names.asp

# input is a command to receive information from the console
# it returns a string, so we need to call a cast function

# int() converts a string to integer, so in this case our variable
# more on casting here, https://www.w3schools.com/python/python_casting.asp

# first and second will be interpreted as integers, becuase they are
# receiving integer types on assignment.
first = int(input("Enter the first positive integer : "))
second = int(input("Enter the second positive integer : "))

# print is a function to print to the console.
# It receives a string an write to the console.

# Here we need to use another cast function, since
# first and second are integers and print expects a string

# we use 2 + commands, one for adding the integers and
# other to concatenate the strings.
print("The sum is : " + str(first + second))
